File: ContosoWebFormsScaffolding/Samples.CS/3_Associations/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace Samples.Associations
{
    public class Product
    {
        public int Id { get; set; }

        [Required(ErrorMessage="Product name is required.")]
        public string Name { get; set; }

        public decimal Price { get; set; }

        // The CategoryId property is scaffolded. It's what
        // gets set in the DropDownList.
        [Display(Name = "Category")]
        public int CategoryId { get; set; }


        // The Category property is the navigation property.
        // This property is not scaffolded but it creates the 
        // association between products and categories.
        public Category Category { get; set; }
    }
}

File: extensibilty-for-asp.net-devs/ContosoWebFormsScaffolding/Samples.VB/3_Associations/Product.vb
Imports System.ComponentModel.DataAnnotations

Public Class Product

    Public Property Id As Integer

    <Required(ErrorMessage:="Product name is required.")>
    Public Property Name As String

    Public Property Price As Decimal

    ' The CategoryId property is scaffolded. It's what
    ' gets set in the DropDownList.
    <Display(Name:="Category")>
    Public Property CategoryId As Integer


    ' The Category property is the navigation property.
    ' This property is not scaffolded but it creates the 
    ' association between products and categories.
    Public Property Category As Category

End Class
